backend/src/main/kotlin/app/ehrenamtskarte/backend/cards/database/repos/CardRepository.kt

```diff
@@ -57,7 +57,7 @@ object CardRepository {
             this.expirationDay = expirationDay
             this.issueDate = Instant.now()
             this.regionId = EntityID(regionId, Regions)
-            this.issuerId = EntityID(issuerId, Administrators)
+            this.issuerId = if (issuerId != null) EntityID(issuerId, Administrators) else null
```

Comment:
:question: would it perhaps make sense to set the `issuerId` to something representing the self-service portal instead of setting it to null, i.e. to introduce a hard-coded "administrator" to identify the self-service portal.

Reply:
I considered this option as an alternative, but was not sure about the benefit

Reply:
Yes, not sure either. I just thought it might prevent accidentally missing the issuerId. But I guess this is only a small benefit.

Reply:
hm the problem is we might need a new role for that because we need a "protected" user that will be created once in this case if we set an issuer id. (i think we should avoid this) i thought about it could make sense to set a flag in the card table like a source (SelfService) (InternalService) or sth meaningful to indicate that the issuerId is not missing by mistake.

Reply:
@f1sh1918 or as I noted in "open questions", maybe we can add a reference column to user entitlements table.
it might also indicate, that it's a self-service.
what do you think?

Reply:
there is a separate task for that. I think we agreed on to keep the card table as it is but create a separate table for that relationship
https://github.com/digitalfabrik/entitlementcard/issues/1424

Reply:
True, I forgot about that, thanks